### PR TITLE
Catch documentation and examples errors

### PR DESCRIPTION
**Pull request by rafmudaf**
# Improve CI configuration to abort when documentation build errors and test for repo dependency in Examples
This pull request adds a setting to the documentation configuration that will stop the build when a notebook within the docs has an error in any code cell. Currently, the documentation is configured to proceed with building the docs even if a notebook errors. This was made clear recently during a FLORIS workshop where errors in the documentation were a surprise because GitHub Actions had not failed.

Another change included here runs the examples in a directory outside of the top-level repository directory. This tests for whether any feature described in the examples is relying on the directory structure of the repository. For example, see #737.

**Note** accepting this pull request means the documentation CI pipeline will fail. This is expected and must be resolved through other fixes to the documentation ASAP. See #733 for more info.

Additionally, I fixed spelling errors in docstrings throughout the software, and addressed a few warnings from Jupyter Book and Sphinx. Other warnings from autodoc and auto summary Sphinx extensions are suppressed since they are a consequence of autogenerating the API documentation.

## Related issue
#733
#737

## Impacted areas of the software
Docs
Automated check for running the examples

### CODE REVIEW

**Comment by paulf81** [comment]
This seems like a good change to make to be made aware of any broken notebooks, support!

File: docs/_config.yml [hunk 1]
```diff
@@ -44,13 +44,17 @@ sphinx:
...
-  - 'sphinx.ext.napoleon'       # Formats google and numpy docstring styles
+  - 'sphinx.ext.napoleon'               # Formats google and numpy docstring styles
   - 'sphinxcontrib.mermaid'
   config:
     html_theme: sphinx_book_theme
     templates_path:
     - '_templates'
     language: 'python'
+    nb_execution_show_tb: True          # Shows the stack trace in stdout; its suppressed otherwise
+    nb_execution_raise_on_error: true   # Stops the Sphinx build if there is an error in a notebook. See https://github.com/executablebooks/jupyter-book/issues/2011
```
[the remark this answers is not in the document]
Comment (RHammond2) [reply docs/_config.yml:55]: Per our conversation a few moments ago, it seems that this is heavily dependent on the version of `jupyter-book` for where these settings should live. However, neither version solved the issue where it doesn't break the build process.

**Comment by rafmudaf** [comment]
@RHammond2 I've pinned the Jupyter Book version to [v0.14.0](https://github.com/executablebooks/jupyter-book/releases/tag/v0.14.0) since that one supports the `nb_execution_show_tb` and `nb_execution_raise_on_error` settings while v0.13 does not. FWIW I couldn't actually verify that in the code that 0.13 does not support those, but I found [this line](https://github.com/executablebooks/jupyter-book/pull/1842/files#diff-f64371f310bcd3bbf6ead5827d929a4eb13b7516806d6658485a996e39eb2556R73) changed in v0.14 and my own testing verifies this.

This change now catches any errors in notebook execution during the documentation build process. You can see the failing [Actions job on my fork](https://github.com/rafmudaf/floris/actions/runs/6870004534/job/18684009117#step:5:1152).

**Review by RHammond2 — APPROVED** [review]
This looks good to me given that the failure is breaking the build now!

**Pull request merged**